pkg/oauth/github/github.go

```diff
@@ -157,11 +168,18 @@ func (c *OAuthClient) decideRole(user string, teams []*github.Team) (role model.
 		return
 	}
 
+	// If the ProjectRBACConfig was not defined or the current user does not belong to it but
+	// the current user belongs to any of the UserGroups, returns role.
+	if len(role.ProjectRbacRoles) != 0 {
+		return
```

Comment:
should we at least add `role.ProjectRole` value here, looks like in this case, there is no value for the `role.ProjectRole` field.

Reply:
@khanhtc1202 
Thanks.

> I think role.ProjectRole field is redundant and could be removed as well
This is to avoid breaking changes.
It's okay that `role.ProjectRole` will be deleted in the future but I think both of them must exist for a while at least.

Reply:
I get your point, but what if we remove this `role.ProjectRole` 🤔 I think users only need to re-login to controlplane,isn't it?

Reply:
> I think users only need to re-login to controlplane,isn't it?

Basically, you're right but we need to fix this stage first if we do so.
And IMO, it might annoy the user because they could not run any API temporarily.
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/service/webservice/service.pb.auth.go

@nghialv 
wdyt?

Reply:
as far as I can see, only webservice uses that auth logic hence piped and pipectl work fine for this change. Users who use pipecd controlplane only be logout and need to re-login, that is not much compared to the pain if we keep maintaining both ProjectRole and ProjectRBACRole ( ref [code](https://github.com/pipe-cd/pipecd/pull/3666/files#diff-1f6c40d1c5acaba04b80af98f46711ed2518cfee37b74c3c87bb4be70a6271b8R36-R39): their definition look overlapped and similar to each other) cause hard to read and hard to maintain as well. So one vote from me to remove it.

Reply:
Thank you for your options.
Let me summarize.

① The way to remove `role.ProjectRole`.
props: 
It's easy to understand and maintain the code.

cons:
The number of diff codes might be more because we have to fix around ProjectRole at once, that's why this might cause some degradations.
This way might annoy the current users because they have to re-login to pass authentication again.

② The way to keep both `role.ProjectRole` and `role.ProjectRBACRole` for a while.
props:
This is safer compared to ①.

cons:
It might be hard to read and maintain the code during both of them exist.

Reply:
> only webservice uses that auth logic hence piped and pipectl work fine for this change. Users who use pipecd controlplane only be logout and need to re-login, that is not much compared to the pain if we keep maintaining both ProjectRole and ProjectRBACRole

I feel the same. If only `re-login` is needed, I prefer to remove it to avoid future pain.

Reply:
umm...
If we do so, I feel it's too many changes because we need to include fixes in this area.
How about adopting ② temporally, and then we remove the old one after checking whether it works?
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/service/webservice/service.pb.auth.go

Reply:
Yes, that strategy is ok to me as well. 👍